### ISSUE TITLE
I have a few doubts about the WeightedFocal(Loss) Class.

### ISSUE THREAD

**Issue by chengfeng0113** [cvcode18/imbalanced_learning#9]
1. The default value of this parameter from_sigmoid is False. If it is False, it means that the sigmoid function is not used to process the network output. When the loss function is executed, the following code is executed:
`if not self._from_sigmoid:
     max_val = F.relu(-pred)
     loss = pred - pred * label + max_val + F.log(F.exp(-max_val) + F.exp(-pred - max_val))`
I don't really understand this code，and it is not mentioned in the paper.
However, this class instance was initialized without setting from_sigmoid to True.
` sigmoid_loss = WeightedFocal()
  attention_loss = AttHistory()`
So I want to know which loss function is used in this paper？

**Comment by chengfeng0113** [comment]
I used pytorch to reproduce this paper, but mAP was only 0.74. I did not use the attention module, only using ResNet101 network training mAP can reach 0.67, plus the attention module can only reach 0.74. I do not know that link has problems.I trained with WIDER dataset。

**Comment by DeepTiming** [comment]
hello? Can you share your pytorch code  which  reproduce to this paper? Thank you very much!!

**Comment by dootoo** [comment]
> hello? Can you share your pytorch code which reproduce to this paper? Thank you very much!!
Sorry, there may be some problems with the code at present, and the accuracy of the author's paper can not be reproduced. It's also being debugged recently. If the accuracy of the paper can be achieved, I will share the code.

**Comment by chengfeng0113** [comment]
> hello? Can you share your pytorch code which reproduce to this paper? Thank you very much!!

Sorry, there may be some problems with the code at present, and the accuracy of the author's paper can not be reproduced. It's also being debugged recently. If the accuracy of the paper can be achieved, I will share the code.

**Comment by DeepTiming** [comment]
Get it! Good lucky!

**Comment by nsarafianos** [comment]
Hi @dootoo 

Apologies for the late response and thank you for your interest in our work. 

1. WIDER: Another person had some problems with reproducing our results but they managed to achieve 83% with just a ResNet101 and 84.7% with an attention module which is in-line with our paper. Please check [here](https://github.com/cvcode18/imbalanced_learning/issues/3#issuecomment-477696407) for some tips that can get you to the desired performance. 

2. PETA: After some back and forth with another person [here](https://github.com/cvcode18/imbalanced_learning/issues/7#issuecomment-468097855) she managed to achieve better results than the ones we report in our paper. 

3. Focal Loss: The loss function we use (wFL) is a small variation of Eq. 5 of the [original paper](https://arxiv.org/pdf/1708.02002.pdf) where the alpha parameter is the class imbalance ratio per attribute in the training set. 

Could you please take a look at these and train a ResNet-101 with normal BCE and check if you can achieve a mAP performance close to the one reported. Then add the attention mechanism and train only these modules. Finally, fine-tune everything together with weighted-FL. 

I'm here for any questions you might have.
